File: src/p0011/java/solution.java
class Solution {
    public int maxArea(int[] hs) {
        int result = 0, i = 0, j = hs.length;
        while (i + 1 < j) {
            if (hs[i] < hs[j - 1]) {
                result = Math.max(result, (j - 1 - i) * hs[i]);
                i++;
            } else {
                result = Math.max(result, (j - 1 - i) * hs[j - 1]);
                j--;
            }
        }
        return result;
    }
}

File: src/p0011/python/solution.py
class Solution:
    def maxArea(self, hs: List[int]) -> int:
        result, i, j = 0, 0, len(hs)
        while i + 1 < j:
            if hs[i] < hs[j - 1]:
                result, i = max(result, (j - 1 - i) * hs[i]), i + 1
            else:
                result, j = max(result, (j - 1 - i) * hs[j - 1]), j - 1
        return result
